### ISSUE TITLE
Tool scripts export categories and their properties to the inspector twice

### ISSUE THREAD

**Issue by MagdielM** [godotengine/godot#63454]
### Godot version

4.0.alpha12.official [2c11e6d9e]

### System information

Windows 10

### Issue description

Tool scripts that export properties under a category using `@export_category` will have this category and properties shown twice in the inspector.
![image](https://user-images.githubusercontent.com/56076033/180844007-75ecb447-2a7f-46fa-b5bc-ee86df5b5287.png)

![image](https://user-images.githubusercontent.com/56076033/180844068-725fb030-7db9-4bfe-9a02-1c5ea4a5924a.png)

### Steps to reproduce

From a new project:

1. Create a tool script and attach it to whichever object it extends.
2. Use the `@export_category` annotation to add a category.
3. Export one or more properties.
4. Inspect the object with the tool script attached.
5. Note how the category and properties are displayed twice.

From the reproduction project:

1. Open `new_resource.tres`.
2. Note how the "Category" category and its associated properties are displayed twice.

### Minimal reproduction project

[Duplicate Property Repro.zip](https://github.com/godotengine/godot/files/9183794/Duplicate.Property.Repro.zip)

**Comment by Calinou** [comment]
cc @YuriSizov

**Comment by YuriSizov** [comment]
This is related to the logic introduced by https://github.com/godotengine/godot/pull/32428. The approach from that PR is based around taking a list of properties that doesn't have script-class grouping and rebuilding that part from scratch, and then removing the original properties.

The problem is, it breaks in the case of annotations because it is made to stop at the first category. So if you add a category in your property list, it thinks it's time to stop there. Accidentally, this doesn't happen if you use the old school way with `_get_property_list()`, but that seems to actually be an issue with the original implementation.

So this works:
```gdscript
@tool
extends Node

func _get_property_list() -> Array:
	var properties := []
	
	properties.push_back({
		"name": "Extra Category",
		"type": TYPE_NIL,
		"usage": PROPERTY_USAGE_CATEGORY,
	})
	properties.push_back({
		"name": "some_variable",
		"type": TYPE_INT,
		"USAGE": PROPERTY_USAGE_DEFAULT,
	})
	
	return properties
```

And this breaks:
```gdscript
@tool
extends Node

@export_category("Extra Category")
@export var some_variable : int
```

-----

While trying to figure out the logic behind https://github.com/godotengine/godot/pull/32428 and if it could be improved, I realized that in the current master `_get_property_list` just doesn't work, unless you have categories. And thus it all comes full circle. Luckily, there is a PR that fixes the `_get_property_list` issues by removing duplicated properties instead of removing everything in the list between two predefined marks. This one: https://github.com/godotengine/godot/pull/58443

I've tested https://github.com/godotengine/godot/pull/58443 and it fixes this issue as well. So let's wait for it to either be merged, or improved.
